source/grid/grid_generator_pipe_junction.cc

```diff
@@ -69,6 +69,194 @@ namespace
+
+      /**
+       * The direction vector perpendicular to both direction and
+       * normal_direction.
+       */
+      const Tensor<1, spacedim> dxn;
+    };
+
+
+
+    template <int dim, int spacedim>
+    Manifold<dim, spacedim>::Manifold(
+      const Tensor<1, spacedim> &normal_direction,
+      const Tensor<1, spacedim> &direction,
+      const Point<spacedim> &    point_on_axis,
+      const AdditionalData &     data,
+      const double               tolerance)
+      : ChartManifold<dim, spacedim, 3>(Tensor<1, 3>({0, 2. * numbers::PI, 0}))
+      , normal_direction(normal_direction)
+      , direction(direction)
+      , point_on_axis(point_on_axis)
+      , data(data)
+      , tolerance(tolerance)
+      , dxn(cross_product_3d(direction, normal_direction))
+    {
+      Assert(spacedim == 3,
+             ExcMessage(
+               "PipeSegment::Manifold can only be used for spacedim==3!"));
+
+      Assert(std::abs(normal_direction.norm() - 1) < tolerance,
+             ExcMessage("Normal direction must be unit vector."));
+      Assert(std::abs(direction.norm() - 1) < tolerance,
+             ExcMessage("Direction must be unit vector."));
+      Assert(normal_direction * direction < tolerance,
+             ExcMessage(
+               "Direction and normal direction must be perpendicular."));
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::unique_ptr<dealii::Manifold<dim, spacedim>>
+    Manifold<dim, spacedim>::clone() const
+    {
+      return std::make_unique<Manifold<dim, spacedim>>(
+        normal_direction, direction, point_on_axis, data, tolerance);
+    }
+
+
+
+    template <int dim, int spacedim>
+    Point<3>
+    Manifold<dim, spacedim>::pull_back(const Point<spacedim> &space_point) const
+    {
+      // First find the projection of the given point to the axis.
+      const Tensor<1, spacedim> normalized_point = space_point - point_on_axis;
+      double                    lambda           = normalized_point * direction;
+      const Point<spacedim>     projection = point_on_axis + direction * lambda;
+      const Tensor<1, spacedim> p_diff     = space_point - projection;
+      const double              r          = p_diff.norm();
+
+      Assert(r > tolerance * data.skeleton_length,
+             ExcMessage(
+               "This class won't handle points on the direction axis."));
+
+      // Then compute the angle between the projection direction and
+      // another vector orthogonal to the direction vector.
+      const double phi =
+        Physics::VectorRelations::signed_angle(normal_direction,
+                                               p_diff,
+                                               /*axis=*/direction);
+
+      // Map the axial coordinate to a cylinder of height one.
+      lambda /= compute_z_expansion(r * std::cos(phi), r * std::sin(phi), data);
+
+      // Return distance from the axis, angle and signed distance on the axis.
+      return {r, phi, lambda};
+    }
```

Comment:
This is in essence what the `Cylinder` manifold probably needs to do, except for an arbitrarily located axis, right?

Reply:
That is exactly what the `Cylinder` manifold does. Basically only line 234 is new. However, with this one additional line, I still thought that adding a dedicated `Manifold` class for this geometry is the way to go here.